docs/source/en/perf_infer_gpu_many.md

```diff
@@ -22,6 +22,73 @@ Note: A multi GPU setup can use the majority of the strategies described in the
+  
+</Tip>
+
+### Decoder models
+
+For text models, especially decoder-based models (GPT, T5, Llama, etc.), the `BetterTransformer` API converts all attention operations to use the [`torch.nn.functional.scaled_dot_product_attention` operator](https://pytorch.org/docs/master/generated/torch.nn.functional.scaled_dot_product_attention) (SDPA) that is only available in PyTorch 2.0 and onwards. 
+
+To convert a model to `BetterTransformer`:
+
+```python
+from transformers import AutoModelForCausalLM
+
+model = AutoModelForCausalLM.from_pretrained("facebook/opt-350m")
+# convert the model to BetterTransformer
+model.to_bettertransformer()
+
+# Use it for training or inference
+```
+
+SDPA can also call [Flash-Attention](https://arxiv.org/abs/2205.14135) kernels under the hood. In order to force the usage of Flash Attention or check that Flash Attention is available in a given setting (hardware, problem size), the context manager [`with torch.backends.cuda.sdp_kernel(enable_flash=True, enable_math=False, enable_mem_efficient=False)`](https://pytorch.org/docs/master/backends.html#torch.backends.cuda.sdp_kernel) can be used:
+
+
+```python
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+tokenizer = AutoTokenizer.from_pretrained("facebook/opt-350m")
+model = AutoModelForCausalLM.from_pretrained("facebook/opt-350m").to("cuda")
+# convert the model to BetterTransformer
+model.to_bettertransformer()
+
+input_text = "Hello my dog is cute and"
+inputs = tokenizer(input_text, return_tensors="pt").to("cuda")
+
+with torch.backends.cuda.sdp_kernel(enable_flash=True, enable_math=False, enable_mem_efficient=False):
+    outputs = model.generate(**inputs)
+
+print(tokenizer.decode(outputs[0], skip_special_tokens=True))
+```
+
+If you see a bug with a traceback saying 
+
+```bash
+RuntimeError: No available kernel.  Aborting execution.
+```
+
+you may try to use the PyTorch nightly version, which may have a larger coverage for Flash Attention:
+
+```bash
+pip3 install -U --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu118
+
+
+Have a look at this [blog post](https://pytorch.org/blog/out-of-the-box-acceleration/) to learn more about what is possible to do with the `BetterTransformer` + SDPA API.
+
+### Encoder models
+
+For encoder models during inference, BetterTransformer will dispatch the forward call of encoder layers to an equivalent of [`torch.nn.TransformerEncoderLayer`](https://pytorch.org/docs/stable/generated/torch.nn.TransformerEncoderLayer.html) that will execute the fast path implementation of the encoder layers.
+
+As `torch.nn.TransformerEncoderLayer` fastpath does not support training, we instead dispatch to `torch.nn.functional.scaled_dot_product_attention` during training, that do not leverage nested tensors but can leverage Flash Attention or Memory-Efficient Attention fused kernels.
```

Comment:
```suggestion
Because `torch.nn.TransformerEncoderLayer` fastpath does not support training, it is dispatched to `torch.nn.functional.scaled_dot_product_attention` instead, which does not leverage nested tensors but can use Flash Attention or Memory-Efficient Attention fused kernels.
```